kahuna/public/js/edits/service.js

```diff
@@ -149,6 +149,17 @@ service.factory('editsService',
         return existingRequestPool.promise;
     }
 
+    function perform(resource, action, originalImage) {
+        const newRequest = resource.perform(action).
+              then(action => resource.get().then(edit => getSynced(originalImage, newImage => matches(edit, newImage))));
+
+        const existingRequestPool = updateRequestPools.get(resource) ||
+            registerUpdateRequest(resource, originalImage);
```

Comment:
[Updated to the way more explicit method name](https://github.com/guardian/grid/commit/50bdf59eb1e4e591bc9f6175357f8e6ab6b185fa) to make sure we don't confuse it with anything else later.